src/components/SettingsBarContext.tsx

```diff
@@ -0,0 +1,53 @@
+import { createContext, ReactNode, useState } from 'react';
+import { FeatureStatus } from '../models';
+
+import api from '../service/api';
+
+export type HistoryBarContextType = {
+  handleSetBar: () => void,
+  handleUpdateFavotites: () => void,
+  handleUpdateHistory: () => void,
+  historyBar: FeatureStatus,
+  favoriteContextState: number,
+  historyContextState: number,
+}
+
+export const HistoryBarContext = createContext<HistoryBarContextType>({
+  handleSetBar: () => { },
+  handleUpdateFavotites: () => { },
+  handleUpdateHistory: () => { },
+  historyBar: FeatureStatus.disabled,
+  favoriteContextState: 0,
+  historyContextState: 0,
+});
+
+type HistoryBarProviderType = {
+  children?: ReactNode
+}
+
+export const HistoryBarProvider = ({ children }: HistoryBarProviderType) => {
+  const [historyBar, setHistoryBar] = useState(api.getStore().historyBar);
+  const [favoriteContextState, setFavoriteContextState] = useState(api.getFavoritsIdList().length);
+  const [historyContextState, setHistoryContextState] = useState(api.getHistoryIdList().length);
+
+  const handleSetBar = () => {
```

Comment:
This function name is really weird. It is impossible to understand what does it mean from name

Reply:
updated